data-structures-queues/take-front.js

```diff
@@ -0,0 +1,8 @@
+/* exported takeFront */
+
+function takeFront(queue) {
+  if (queue.peek() === undefined) {
+    return undefined;
+  }
+  return queue.dequeue();
```

Comment:
Here you can just have the `return` since `dequeue` will also return `undefined` if the queue is empty.
```JAVASCRIPT
function takeFront(queue) {
  return queue.dequeue();
}
```